rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -160,22 +168,22 @@ public boolean validate(Block block, BlockResult result) {
      * Execute a block, from initial state, returning the final state data.
      *
      * @param block        A block to validate
-     * @param stateRoot    Initial state hash
+     * @param parent       The parent of the block to validate
      * @return BlockResult with the final state data.
      */
-    public BlockResult execute(Block block, byte[] stateRoot, boolean discardInvalidTxs) {
-        return execute(block, stateRoot, discardInvalidTxs, false);
+    public BlockResult execute(Block block, Block parent, boolean discardInvalidTxs) {
```

Comment:
It is really necessary to switch from the stateRoot to the parent block here ? In the future nodes will fast sync, and therefore the first block to be executed won't have a block parent to reference. It may have a BlockHeader parent, but not a full block.

Reply:
I think passwing the BlockHeader could suffice. If not, then maybe we can pass both the parent block id and the stateroot, and the code uses one or the other (the one that is != null)

Reply:
I passed the `BlockHeader`. Good catch!